src/stream_engine.cpp

```diff
@@ -784,19 +784,6 @@ int zmq::stream_engine_t::next_handshake_command (msg_t *msg_)
 
         if (rc == 0)
             msg_->set_flags (msg_t::command);
-#ifdef ZMQ_BUILD_DRAFT_API
```

Comment:
`mechanism->status () == mechanism_t::error` becomes true asynchronously in the IO thread that processes the ZAP reply. Here, no useful error code is available. So, I moved emitting such events to the point where the error is identified. 
This also changes from "edge-triggering" to "level-triggering", which might emit multiple events for the same error condition.